h2/settings.py

```diff
@@ -254,12 +254,18 @@ def __len__(self):
         return len(self._settings)
 
 
-    def __eq__(self,other):
-        return self._settings==other._settings
-
-    def __ne__(self,other):
-        return not self._settings==other._settings
-
+    def __eq__(self, other):
+        if isinstance(other, Settings):
+            return self._settings == other._settings
+        else:
+            return NotImplemented
```

Comment:
Sorry @SethMichaelLarson, but again I disagree. =) Per the documentation, [`NotImplemented`](https://docs.python.org/2/library/constants.html#NotImplemented) should be returned when we have no method of determining equality with respect to the other type. This is to allow *other* rich comparison methods to potentially be invoked that may know how to do an equality comparison.

The correct thing to do here is to return `NotImplemented` when we don't have matching types.